backend/schema/validate.go

```diff
@@ -326,6 +331,65 @@ func ValidateModule(module *Module) error {
 	return errors.Join(merr...)
 }
 
+func findDuplicateDecls(decls []Decl) []error {
+	merr := []error{}
+
+	configs := make(map[string][]int)
+	secrets := make(map[string][]int)
+	dbs := make(map[string]int)
+
+	for i, d := range decls {
+		c, ok := d.(*Config)
```

Comment:
Use a type switch here instead of these separate individual type conversions:

```go
switch d := d.(type) {
case *Config:
}
```

Reply:
whoops thank you!